src/reporting/clients/ODataClient.ts

```diff
@@ -0,0 +1,144 @@
+*--------------------------------------------------------------------------------------------*/
+import type { AccessToken } from "@itwin/core-bentley";
+import { RequiredError } from "../interfaces/Errors";
+import type { PagedResponseLinks } from "../interfaces/Links";
+import { ODataEntityResponse, ODataEntityValue, ODataItem, ODataMetaDataEntitySet, ODataMetaDataEntityType, ODataMetaDataProperty, ODataMetaDataSchema, ODataResponse, ODataTable } from "../interfaces/OData";
+import type { EntityListIterator } from "../iterators/EntityListIterator";
+import { EntityListIteratorImpl } from "../iterators/EntityListIteratorImpl";
+import { Collection, getEntityCollectionPage } from "../iterators/IteratorUtil";
+import { OperationsBase } from "../OperationsBase";
+import type { IOdataClient } from "./IODataClient";
+import { XMLParser } from "fast-xml-parser";
+
+export class ODataClient extends OperationsBase implements IOdataClient{
+  public async getODataReport(accessToken: AccessToken, reportId: string): Promise<ODataResponse> {
+    const url = `${this.basePath}/odata/${encodeURIComponent(reportId)}`;
+    const requestOptions: RequestInit = this.createRequest("GET", accessToken);
+    return this.fetchJSON<ODataResponse>(url, requestOptions);
+  }
+
+  public async getODataReportEntityPage(accessToken: AccessToken, reportId: string, odataItem: ODataItem, sequence: number): Promise<ODataEntityResponse> {
+    const segments = odataItem.url.split("/");  // region, manifestId, entityType
+    if (segments.length !== 3) {
+      throw new RequiredError(
+        "odataItem",
+        "odata item was invalid.",
+      );
+    }
+    const url = `${this.basePath}/odata/${encodeURIComponent(reportId)}/${odataItem.url}?sequence=${encodeURIComponent(sequence)}`;
+    const requestOptions: RequestInit = this.createRequest("GET", accessToken);
+    return this.fetchJSON(url, requestOptions);
+  }
+
+  public async getODataReportEntities(accessToken: AccessToken, reportId: string, odataItem: ODataItem): Promise<Array<ODataEntityValue>> {
+    const segments = odataItem.url.split("/");  // region, manifestId, entityType
+    if (segments.length !== 3) {
+      throw new RequiredError(
+        "odataItem",
+        "odata item was invalid.",
+      );
+    }
+    const reportData: Array<ODataEntityValue> = [];
+    const oDataReportEntitiesIt = this.getODataReportEntitiesIterator(accessToken, reportId, odataItem);
+    for await(const oDataReportEntity of oDataReportEntitiesIt) {
+      reportData.push(oDataReportEntity);
+    }
+    return reportData;
+  }
+
+  public getODataReportEntitiesIterator(accessToken: AccessToken, reportId: string, odataItem: ODataItem): EntityListIterator<ODataEntityValue> {
+    const segments = odataItem.url.split("/");  // region, manifestId, entityType
+    if (segments.length !== 3) {
+      throw new RequiredError(
+        "odataItem",
+        "odata item was invalid.",
```

Comment:
These error messages are inconsistent with the ones that are in other clients. Make it consistent and capitalized.

Reply:
changed